pkg/utils/utils.go

```diff
@@ -0,0 +1,47 @@
+package utils
+
+import (
+	"fmt"
+	"os"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// NewClientSet returns a clientset using in cluster config.
+func NewClientSet() (*kubernetes.Clientset, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error creating in-cluster config: %w", err)
+	}
+	config.ContentType = runtime.ContentTypeProtobuf
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("error creating clientset: %w", err)
```

Comment:
nit: don't preface with "error" or similar. With that removed, "create" sounds better and is more concise.

Reply:
Fixed this and a few other places.